File: 6function_overriding.py
class Animal:
    def sound(self):
        print("Animal makes a sound")

class Dog():
    def sound(self):
        print("Woof! Woof!")

class Cat():
    def sound(self):
        print("Meow! Meow!")


anim = Animal()
dog = Dog()
cat = Cat()

anim.sound()
dog.sound()  
cat.sound()

File: 6function_overriding.cpp
#include<bits/stdc++.h>
using namespace std;


class Animal{
    public:
        void sound(){
            cout<<"Animla can Make Sound"<<endl;
        }
};

class Cat{
    public:
        void sound(){
            cout<<"Meow! Meow!"<<endl;
        }
};

class Dog{
    public:
        void sound(){
            cout<<"Woof! Woof!"<<endl;
        }
};

int main(){
    
    Animal anim;
    Cat cat;
    Dog dog;
    anim.sound();
    cat.sound();
    dog.sound();
    return 0;
}